### DISCUSSION TITLE
How to call fixtures in non pytest framework

### DISCUSSION THREAD

**Discussion by chetana271285** [pytest-dev/pytest#9501] [Q&A]
I have an in house test framework and I am trying to migrate it to Pytest. My new code has fixtures and test functions. But to migrate the existing classes to pytest framework, how can I call the fixture and return the value to the inhouse framework class.

```python
# old : class1.py: Not using pytest framework
Path= '/tmp/abc.txt'
def get_test_data_dir():
     # opens a file from path and returns the test_data_dir 
class ReadData:
      def __init__(self):
           ## there is proc here that reads the path from config file
           self.test_data_dir = get_test_data_dir()
```

```python
# Another class2.py: using Pytest framework
# conftest.py
@pytest.fixture(scope='session' ,autouse=True)
def load_config(request):
    param1 = request.config.getoption("--test_config")
    param2 = request.config.getoption("--cloud_config")
    return ConfigManager(param1,param2)

@pytest.fixture(scope='session')   
def read_data(load_config):
    return load_config.test_data
```

Now I would like to apply the `read_data` fixture cached value to old class1.py self.test_data_dir if called from pytest. Is there  a way to do this?

**Comment by nicoddemus** [comment]
Hi @chetana271285,

We will need more details:

1. Who runs the tests for the in-house framework? pytest, or the framework has its own runner? Do
2. What's the final objective? To have both co-existing, or eventually replacing the in-house framework with pytest?

**Reply by chetana271285** [reply]
Hi Nicoddemus,

Thank you for you reply.
Currently, the in house framework has its own runner to run the tests and these tests are not modular . 
Example:  To test one feature, the test framework  runs all the other features tests as well which are not needed.  I thought I will make it modular using pytest so that single feature  test can be run when needed.
Goal is to eventually replace the in-house framework tests with pytest.

Thanks,
Chetana

**Reply by nicoddemus** [reply]
> Currently, the in house framework has its own runner to run the tests and these tests are not modular .

Hmm you mean you have a high-level script which calls into your in house framework, and then pytest?

```python
import pytest
import in_house_fw

in_house_fw.main(["src"])
pytest.main(["src"])
```

In that case I don't think it is really possible to share the fixture information between them. Perhaps an alternative would be to write a plugin to enable pytest to run the in-house framework's tests? This would allow to eventually bridge the gap between the two.

**Reply by chetana271285** [reply]
Can you share any examples here ? Because I did not understand pytest.main(["src"]) >> what does the list have here? 
Any examples for pytest plugin as well?

Appreciate your responses!!

**Reply by nicoddemus** [reply]
>  Because I did not understand pytest.main(["src"]) >> what does the list have here?

Ahh my bad, should have been `pytest.main("src")` or something to that effect. The point was that you are invoking `pytest.main` (right?).

> Any examples for pytest plugin as well?

For an example you can look at [pytest-cpp](https://github.com/pytest-dev/pytest-cpp), which collects and runs C++ tests. It implements all the mechanisms required to integrate with pytest: collection, run tests, reports.

**Comment by chetana271285** [comment]
Hi,

I am writing the tests in pytest that calls the inhouse framework ( I'm
trying to get rid of the high level script that calls the inhouse framework
). However I am stuck with one thing here.
As per current framework, the setup configuration has config.yaml and test
cases has testcases.yaml files. The config files are read inside the
inhouse framework.
I am trying to change this to fixtures so that they will be available for
all the tests. But the problem is how do I pass the fixtures to the inhouse
framework? I am trying to parametrize the inhouse framework classes so that
I call them from test methods(which have access to fixtures) . Is it the
right way or is there any better way ?

**Reply by nicoddemus** [reply]
Please see my answer in https://github.com/pytest-dev/pytest/discussions/9501#discussioncomment-1957704
